File: bench/ruby/sieve.rb
# Sieve of Eratosthenes to N, counting primes. Checksum = count of primes <= N.
n = (ENV["BENCH_N"] || "1000000").to_i
comp = Array.new(n + 1, false)
p = 2
while p * p <= n
  unless comp[p]
    j = p * p
    while j <= n
      comp[j] = true
      j += p
    end
  end
  p += 1
end
count = 0
(2..n).each { |k| count += 1 unless comp[k] }
puts count

File: bench/go/sieve.go
// Sieve of Eratosthenes to N, counting primes. Checksum = count of primes <= N.
package main

import "fmt"

func main() {
	n := benchN(1000000)
	comp := make([]bool, n+1)
	for p := 2; p*p <= n; p++ {
		if !comp[p] {
			for j := p * p; j <= n; j += p {
				comp[j] = true
			}
		}
	}
	count := 0
	for k := 2; k <= n; k++ {
		if !comp[k] {
			count++
		}
	}
	fmt.Println(count)
}
